assets/js/common/OperationModals/SaptuneSolutionOperationModal.jsx

```diff
@@ -54,25 +56,39 @@ function SaptuneSolutionOperationModal({
   const [checked, setChecked] = useState(false);
   const [solution, setSolution] = useState(NOT_SELECTED);
 
-  const availableSolutions = filter(solutions, ({ available }) =>
-    available(isHanaRunning, isAppRunning)
-  );
+  const availableSolutions = pipe(
+    filter(({ available }) => available(isHanaRunning, isAppRunning)),
+    map((solutionOption) => ({
+      ...solutionOption,
+      disabled: solutionOption.value === currentlyApplied,
+    }))
+  )(solutions);
+
+  const resetSolution = () => setSolution(currentlyApplied || NOT_SELECTED);
+
+  useEffect(() => {
```

Comment:
I'm not sure if we should have this `useEffect`. Here, if the solution changes for example from `null` to something else, the modal updates, but the whole concept changes as well, as we the user might not even notice that the scenario is different, as we maybe are not in a position to apply, and we are in a position to change.

Reply:
That is a fair point I was fighting against as well.
For simplicity I agree that we can avoid such reactiveness.
Thanks